File: mar_25/R/page2.R
# step 1: load the data
df = read.csv('/Volumes/Data/Sunbeam/2019/Feb/DBDA/stats_R/dataset/data\ 1/Social_Network_Ads.csv')

# step 2: clean the data
df$User.ID = NULL
df$Gender = NULL

library(caTools)
results = sample.split(df$Purchased, SplitRatio = 0.8)
df.train = df[results == TRUE, ]
df.test = df[results == FALSE, ]

df$Purchased = factor(df$Purchased)

# step 3: prepare the model/formula/equation
classifier = glm(Purchased ~ ., data = df.train, family = 'binomial')
summary(classifier)

# step 4: predict the value(s)
predictions = predict(classifier, newdata = df.test)

# classify the records
predictions = ifelse(predictions >= 0, 1, 0)

# step 5: test the result
cm = table(df.test$Purchased, predictions)

# step 6: visualize the result66yy6

File: mar_25/python/page2.py
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# step 1: load the data
df = pd.read_csv('/Volumes/Data/Sunbeam/2019/Feb/DBDA/stats_R/dataset/data 1/Social_Network_Ads.csv')
x = df.iloc[:, 2:4].values
y = df.iloc[:, 4].values

# step 2: clean the data

from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=343238)

# step 3: prepare the model/formula/equation
from sklearn.linear_model import LogisticRegression
classifier = LogisticRegression()
classifier = classifier.fit(x_train, y_train)

# step 4: predict the value(s)
predictions = classifier.predict(x_test)
print(predictions)

# step 5: test the result
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test, predictions)
print(cm)


# step 6: visualize the result